common/FileOps.h

```diff
@@ -10,6 +10,7 @@ namespace sorbet {
 class FileOps final {
 public:
     static bool exists(std::string_view filename);
+    static bool matchIsFolderOrFile(std::string_view path, std::string_view ignorePattern, const int pos);
```

Comment:
I didn't understand what this function does from the name.
It also seems niche and is used only by a single file. Maybe move it there?